test/llvm/parallel_loop_access/different_numbers.chpl

```diff
@@ -0,0 +1,53 @@
+proc start_loop1() {}
+proc end_loop1() {}
+
+proc start_loop2() {}
+proc end_loop2() {}
+
+proc start_loop3() {}
+proc end_loop3() {}
+
+//Check whether we generate different metadata number for loops
+proc loop (A, B, n) {
+  for i in vectorizeOnly(1..n) {
+    //CHECK-LABEL: start_loop1
+    start_loop1();
+
+    //CHECK: llvm.mem.parallel_loop_access ![[LOOP1:[0-9]+]]
+    A[i] = 3*B[i];
+
+    //CHECK-LABEL: end_loop1
+    end_loop1();
+  }
+
+  for i in vectorizeOnly(1..n) {
+    //CHECK-LABEL: loop2
+    start_loop2();
+
+    //CHECK: llvm.mem.parallel_loop_access ![[LOOP2:[0-9]+]]
+    //CHECK-NOT: llvm.mem.parallel_loop_access ![[LOOP1]]
+    A[i] = 5*B[i];
+    for j in vectorizeOnly(1..n) {
+      //CHECK-LABEL: start_loop3
+      start_loop3();
+
+      //CHECK: llvm.mem.parallel_loop_access ![[LOOP3:[0-9]+]]
+      //CHECK-NOT: llvm.mem.parallel_loop_access ![[LOOP2]]
```

Comment:
I realized that it is probably not what we want by default. Here is one example of nested loops which are both parallel, but aren't a group of parallel loops:
```chpl
for i in vectorizeOnly(1..n)
{
  A[i] = A[i]*B[i];
  for j in vectorizeOnly(A[i], B[i])
  {
    C[j] = 3*D[j];
  }
}
```
There is a dependency, because result of inner loops depends on multiplication `A[i] = A[i]*B[i]`. It's not a big deal to add this anyway, but keep in mind that according to example in documentation, only the innermost loop has metadata referring to the group of loops, meaning that this test is still valid for this case. My example might not be the best but in most general case the innermost loop might not be 'group parallel' and we shouldn't be marking code as such. I'm unable to test whether this will make difference, because as far as I know llvm currently does not use 'group parallel' metadata for anything.